lib/src/source/hosted.dart

```diff
@@ -1100,3 +1114,95 @@ class _RefAndCache {
   @override
   bool operator ==(Object other) => other is _RefAndCache && other.ref == ref;
 }
+
+extension GCSChecksumValidation on http.StreamedResponse {
+  static const checksumsHeaderName = 'x-goog-hash';
+
+  bool get isFromGCS {
+    // TODO: Does this suffice as a heuristic?
+    return headers.containsKey(checksumsHeaderName);
```

Comment:
This is run after the request is already fired. It could be moved pre-request if we figure out a way to properly detect whether a request is GCS bound in the other TODO.